test/suite/test_truncate12.py

```diff
@@ -110,7 +110,7 @@ def expectNone(lo, hi):
         expectNone(keep + 1, nrows + 1)
 
     def test_truncate12(self):
-        nrows = 5000
+        nrows = 10000
```

Comment:
At least one (I think this one, but I'm not sure) was because the test didn't work on VLCS without more rows.

Usually the problem is with FLCS and it's because by default you get more than 10000 rows on a single FLCS page, so the workload fits on a single page and that violates assumptions, or interacts badly with eviction, or you end up leaving that one page pinned while trying to do something else, and the test doesn't run as expected. In the case of truncate, if there aren't enough pages there may be nothing to fast-truncate, and then the test complains that it didn't actually fast-truncate anything.

In this case (or at least one of them) the problem was that because VLCS does RLE encoding for sequences of duplicate values, and most of the Python tests produce sequences of duplicate values unless someone took steps to avoid it, it needed more rows to generate enough pages for the test to work as intended.

This behavior is unnatural, that is, real workloads are not normally composed of entirely one value, and it's possible we should in general take more steps in the test suite to avoid it.

Reply:
Oh, that makes sense. The thing about FLCS having 10K rows on a page wasn't obvious to me. Good to know, thanks!